core/domain/translation_domain.py

```diff
@@ -18,9 +18,262 @@
+
+
+class EntityTranslation:
+    """The EntityTranslation represents an EntityTranslationsModel for a given
+    version of an entity in a given language.
+
+    Args:
+        entity_id: str. The id of the corresponding entity.
+        entity_type: TranslatableEntityType. The type
+            of the corresponding entity.
+        entity_version: str. The version of the corresponding entity.
+        language_code: str. The language code for the corresponding entity.
+        translations: dict(str, TranslatedContent). The translated content.
+    """
+
+    def __init__(
+        self,
+        entity_id: str,
+        entity_type: feconf.TranslatableEntityType,
+        entity_version: int,
+        language_code: str,
+        translations: Dict[str, TranslatedContent]
+    ):
+        self.entity_id = entity_id
+        self.entity_type = entity_type.value
+        self.entity_version = entity_version
+        self.language_code = language_code
+        self.translations = translations
+
+
+class TranslatableContent:
+    """TranslatableContent represents a content of a translatable object which
+    can be translated into multiple languages.
+
+    Args:
+        content_id: str. The id of the corresponding content.
+        content: ContentInTranslatableContent. The content which can be
+            translated.
+        content_type: TranslatableContentFormat. The type of the
+            corresponding content.
+    """
+
+    def __init__(
+        self,
+        content_id: str,
+        content: feconf.ContentInTranslatableContent,
+        content_type: TranslatableContentFormat
+    ) -> None:
+        self.content_id = content_id
+        self.content = content
+        self.content_type = content_type
+
+    @classmethod
+    def from_dict(
+        cls,
+        translatable_content_dict: TranslatableContentDict
+    ) -> TranslatableContent:
+        """Returns a TranslatableContent object from its dict representation.
+
+        Args:
+            translatable_content_dict: dict. Dict representation of
+                TranslatableContent object.
+
+        Returns:
+            TranslatableContent. The TranslatableContent object.
+        """
+        return cls(
+            translatable_content_dict['content_id'],
+            translatable_content_dict['content'],
+            translatable_content_dict['content_type'])
+
+    def to_dict(self) -> TranslatableContentDict:
+        """Returns the dict representation of TranslatableContent object.
+
+        Returns:
+            dict. The dict representation of TranslatableContent object.
+        """
+        return {
+            'content_id': self.content_id,
+            'content': self.content,
+            'content_type': self.content_type
+        }
+
+
+class TranslatedContent:
+    """Represents the translated content of the TranslatableContent object.
+
+    Args:
+        content: ContentInTranslatableContent. Represents already-translated
```

Comment:
Could you clarify this to explain whether it is storing the content in the original language, or the translation of the content?

Reply:
It is storing the translation of the content.
(_Also updated the docstring as per the above review suggestion._)